File: tests/vb/vb_collections_advanced/list_clear.vb
' vybe-test: vb/vb_collections_advanced/list_clear
' origin: languages/vb/tests/vb/test_vb_collections_advanced.rs

' Vybe test harness — Visual Basic.
'
' Real VB source alongside harness/go/check.go and harness/js/check.js, the way
' test262's assert.js is JavaScript.
'
' A test's verdict is its EXIT CODE. __Check prints its diagnostic BEFORE
' throwing: an uncaught exception surfaces as `RuntimeError: [object]`, which
' says nothing at all.

Module VybeCheck
    Sub __Check(got As String, want As String)
        If got <> want Then
            Console.WriteLine("FAIL: want [" & want & "] got [" & got & "]")
            Throw New Exception("assertion failed")
        End If
    End Sub
End Module

Module M
    Sub Main()
        Dim items As New List(Of Integer)
        items.Add(1)
        items.Add(2)
        items.Add(3)
        __Check(CStr(items.Count), "3")
        items.Clear()
        __Check(CStr(items.Count), "0")
    End Sub
End Module

File: tests/csharp/csharp_collections/list_clear.cs
// vybe-test: csharp/csharp_collections/list_clear
// origin: languages/csharp/tests/csharp/test_csharp_collections.rs

using static __Harness;
using System.Collections.Generic;

var list = new List<int>();
list.Add(1);
list.Add(2);
__P((list.Count).ToString());
list.Clear();
__P((list.Count).ToString());
__Check("2\n0");

public static class __Harness {
    public static string __buf = "";
    public static void __P(string s) { __buf = __buf + s + "\n"; }
    public static void __Pr(string s) { __buf = __buf + s; }
    public static void __Check(string want) {
        if (__buf != want && __buf != want + "\n") {
            Console.WriteLine("FAIL: want [" + want + "] got [" + __buf + "]");
            throw new Exception("assertion failed");
        }
    }
}
